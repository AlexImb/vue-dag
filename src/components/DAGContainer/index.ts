import DAGContainer from './DAGContainer.vue';

export { DAGContainer };
