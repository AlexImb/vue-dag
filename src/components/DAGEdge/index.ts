import DAGEdge from './DAGEdge.vue';

export { DAGEdge };
