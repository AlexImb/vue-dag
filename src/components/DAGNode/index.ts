import DAGNode from './DAGNode.vue';

export { DAGNode };
