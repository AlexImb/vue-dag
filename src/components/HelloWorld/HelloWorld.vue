<template>
  <div>
    <h1>Hello {{ msg }}</h1>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class HelloWorld extends Vue {
  @Prop() private msg!: string;
}
</script>
