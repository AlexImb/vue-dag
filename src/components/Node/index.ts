import Node from './Node.vue';

export { Node };
