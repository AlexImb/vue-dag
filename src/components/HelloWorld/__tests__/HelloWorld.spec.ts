import { shallowMount, Wrapper } from '@vue/test-utils';
import { HelloWorld } from '@/components/HelloWorld';

let wrapper: Wrapper<HelloWorld>;

describe('HelloWorld', () => {
  beforeEach(() => {
    wrapper = shallowMount(HelloWorld);
  });

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders props.msg when passed', () => {
    const msg = ' test!';
    wrapper = shallowMount(HelloWorld, {
      propsData: { msg },
    });
    expect(wrapper.text()).toMatch(msg);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
