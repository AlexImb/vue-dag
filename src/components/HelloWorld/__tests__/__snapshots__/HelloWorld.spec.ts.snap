// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld render correctly 1`] = `
<div>
  <h1>Hello </h1>
</div>
`;

exports[`HelloWorld renders props.msg when passed 1`] = `
<div>
  <h1>Hello test!</h1>
</div>
`;
